sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -309,7 +360,8 @@ def on_merge(self, to_be_merged, merge_result, context):
         self.early.on_merge(
             to_be_merged, merge_result, NestedContext(context, 'early'))
 
-  def should_fire(self, watermark, window, context):
+  def should_fire(self, watermark, window, context,
+                  time_domain=TimeDomain.WATERMARK):
     if self.is_late(context):
       return self.late.should_fire(
           watermark, window, NestedContext(context, 'late'))
```

Comment:
<!--new_thread; commit:a68ceca8046b6d42f941aa4016a528ec2ee3808a; resolved:0-->
`should_fire()` should take extra `time_domain` argument.

Reply:
done